projects/arlas-components/src/lib/components/results/result-actions/result-actions.component.ts

```diff
@@ -157,18 +156,56 @@ export class ResultActionsComponent implements OnInit, OnChanges, OnDestroy {
             reverseAction: action.reverseAction,
             icon: action.icon,
             fields: action.fields,
+            filters: action.filters,
             show: action.show
           });
         });
         this.actions = item.actions;
         this.updateActions();
       });
-    } else if (item && item.actions && item.actions.length > 0) {
+    } else if (item?.actions?.length > 0) {
       this.actions = item.actions;
       this.updateActions();
     }
   }
 
+  /**
+   * Update the visibility of actions after the item has been hovered
+   */
+  private onItemHovered() {
+    /** Always show non reversible actions. */
+    this.actions.filter(a => !ActionHandler.isReversible(a)).forEach(a => a.show = true);
+    /** We check if reversible actions has 'fields'.
+     * - If one of the fields values is absent in the current item, the action will be hidden. */
+    this.actions.filter(a => ActionHandler.isReversible(a) && a.show === undefined).forEach(a => {
+      if (a.fields) {
+        this.detailedDataRetriever.getValues(this.item.identifier, a.fields).pipe(
+          take(1)).subscribe(values => a.show = values.filter(v => !v).length === 0);
+      } else if (a.filters) {
+        this.detailedDataRetriever.getMatch(this.item.identifier, a.filters).pipe(
+          take(1)).subscribe(v => {
+            a.matched = v.matched;
```

Comment:
Maybe we can add a security check here where the `a.filters.length`, should be equal to `a.matched.length`